python/az/aro/azext_aro/custom.py

```diff
@@ -128,9 +135,42 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
                        parameters=oc)
 
 
-def aro_delete(client, resource_group_name, resource_name, no_wait=False):
+def aro_delete(cmd, client, resource_group_name, resource_name, no_wait=False):
     # TODO: clean up rbac
 
+    try:
+        oc = client.get(resource_group_name, resource_name)
+
+        master_subnet = oc.master_profile.subnet_id
+        worker_subnets = {w.subnet_id for w in oc.worker_profiles}
+
+        master_parts = parse_resource_id(master_subnet)
+        vnet = resource_id(
+            subscription=master_parts['subscription'],
+            resource_group=master_parts['resource_group'],
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=master_parts['name'],
+        )
+
+        aad = AADManager(cmd.cli_ctx)
+        rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
+        rp_client_sp = aad.get_service_principal(rp_client_id)
+
+        # Customers frequently remove the RP's permissions, then cannot
+        # delete the cluster. Where possible, fix this before attempting
+        # deletion.
+        if rp_client_sp:
+            sp_id = rp_client_sp.object_id
+            assign_contributor_to_vnet(cmd.cli_ctx, vnet, sp_id)
+            assign_contributor_to_routetable(cmd.cli_ctx,
+                                             worker_subnets | {master_subnet},
+                                             sp_id)
+    except (CloudError, HttpOperationError) as e:
+        # Default to old deletion behaviour in case operations throw an
+        # exception above.
+        pass
```

Comment:
I'd like to see the error logged in such a way that there is *some* way to see it if we need to (--debug, whatever) and that it doesn't worry people and cause them to ask for support.